jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/ApplyService.java

```diff
@@ -483,27 +470,26 @@ private void doCreateCustomResourceDefinition(CustomResourceDefinition entity, S
         }
     }
 
-    private void applyCustomResource(CustomResourceDefinitionContext context, GenericCustomResource genericCustomResource, String sourceName)
-        throws IOException {
-
-        String name = genericCustomResource.getMetadata().getName();
-        String applyNamespace = applicableNamespace(genericCustomResource, namespace, fallbackNamespace);
-        String apiGroupWithKind = KubernetesHelper.getFullyQualifiedApiGroupWithKind(context);
-        Objects.requireNonNull(name, "No name for " + genericCustomResource + " " + sourceName);
+    private void applyCustomResource(GenericKubernetesResource genericKubernetesResource, String sourceName) {
```

Comment:
Why are we keeping this as a separate method?

`public void applyGenericKubernetesResource(GenericKubernetesResource dto, String sourceName)` is just calling this one, we should only keep one of the two (probably the one that was already exposed/public)